mopidy/config/types.py

```diff
@@ -178,30 +219,117 @@ def serialize(self, value, display=False):
             raise ValueError(f"{value!r} is not a boolean")
 
 
+class Pair(ConfigValue):
+    """Pair value
+
+    The value is expected to be a pair of elements, separated by a specified delimiter.
+    Values can optionally not be a pair, in which case the whole input is provided for
+    both sides of the value.
+    """
+
+    def __init__(
+        self, optional=False, optional_pair=False, separator="|", subtypes=None
+    ):
+        self._required = not optional
+        self._optional_pair = optional_pair
+        self._separator = separator
+        if subtypes:
+            self._subtypes = subtypes
+        else:
+            self._subtypes = (String(), String())
+
+    def deserialize(self, value):
+        raw_value = decode(value).strip()
+        validators.validate_required(raw_value, self._required)
+        if not raw_value:
+            return None
+
+        if self._separator in raw_value:
+            values = raw_value.split(self._separator, 1)
+        elif self._optional_pair:
+            values = (raw_value, raw_value)
+        else:
+            raise ValueError("must have separator")
```

Comment:
I guess this might hit end users, so it should be even more explicit:

```suggestion
            raise ValueError(f"Config value must include {self._separator!r} separator: {raw_value}")
```

Reply:
I've made this change :+1: